### PR TITLE
ci: expand lighthouse usage and reporting

### PR DESCRIPTION
**Pull request by Westbrook**
## Description
- build the test URL, rather than extract it from other functionality
- add a comparison to `main`
- [x] report the results into the PR
- [x] possibly compare to `latest`
- go from there

## How has this been tested?
-   [ ] _Test case 1_
    1. At some point the lighthouse results pre/post a PR should display in a comment below...

## Types of changes
-   [x] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist
-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

### CODE REVIEW

File: .github/lighthouse/lighthouse-config.js [hunk 1]
```diff
@@ -2,12 +2,11 @@ export default {
     extends: 'lighthouse:default',
     settings: {
         onlyCategories: [
-            // Excluded categories do not currently pass
-            // "accessibility",
-            // "best-practices",
+            'accessibility',
```
Comment (Westbrook) [inline .github/lighthouse/lighthouse-config.js:5]: Turning on the categories here does not seem to turn on the audits. This way we can report their scores to the PR.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -41,7 +41,7 @@ jobs:
                         repo: context.repo.repo,
                         issue_number: context.issue.number,
                       }).then(({data}) => {
-                        const priorComment = data.find(comment => comment.body.startsWith('# Branch Preview'));
+                        const priorComment = data.find(comment => comment.body.startsWith('## Branch preview') || comment.body.startsWith('# Branch Preview'));
```
Comment (Westbrook) [inline .github/workflows/test.yml:44]: This messages out the older version from existing PRs by replacing them or the new versions with the updated comment.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -78,27 +78,52 @@ jobs:
...
-                  script: |
-                      const { buildPreviewURLComment } = await import('${{ github.workspace }}/tasks/build-preview-urls-comment.js');
-                      const body = buildPreviewURLComment(process.env.GITHUB_HEAD_REF);
-                      const urlRegex = /(https?:\/\/[^\s]+)/g;
-
-                      const docsURL = body.match(urlRegex)?.[0];
-                      core.setOutput('docsURL', docsURL);
+            - name: Extract branch name
+              shell: bash
+              run: echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_OUTPUT
```
Comment (Westbrook) [inline .github/workflows/test.yml:83]: This still feels like too much for for GitHub to be able to tell us the branch, but it's less than reaching into an unrelated script.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -78,27 +78,52 @@ jobs:
...
-                  uploadArtifacts: true
+                  runs: 3
                   temporaryPublicStorage: true
+                  uploadArtifacts: true
+                  urls: |
+                      https://opensource.adobe.com/spectrum-web-components/
+                      https://main--spectrum-web-components.netlify.app/
+                      https://${{ steps.extract_branch.outputs.branch }}--spectrum-web-components.netlify.app/
+
+            - name: Post Results
```
Comment (Westbrook) [inline .github/workflows/test.yml:99]: This is the third time we've using this style of job in the GH CI, totally worth abstracting, but 100% not something I'm going to spend time on at this juncture.

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -78,27 +78,52 @@ jobs:
...
+              uses: actions/github-script@v6
+              with:
+                  script: |
+                      const { buildLighthouseComment } = await import('${{ github.workspace }}/tasks/build-lighthouse-comment.js');
+                      const body = buildLighthouseComment(${{ steps.lighthouse.outputs.links }}, ${{ steps.lighthouse.outputs.manifest }});
+                      github.rest.issues.listComments({
```
Comment (hunterloftis) [inline .github/workflows/test.yml:105]: I'm not sure what context this runs in... would `const data = await github.rest.issues.listComments(...)` work?

**Review by hunterloftis — APPROVED** [review]

File: .github/workflows/test.yml (hunk 5 above)
Comment (Westbrook) [reply .github/workflows/test.yml:105]: It's no longer recent, but with the post initial creation move to `await import()` above, we should have access to that here and be able to flatten out the asynchrony here. I'll put this and the abstraction process into an issue to track and clarify steps. Great call!

**Pull request merged**